runtime/compiler/p/codegen/PPCPrivateLinkage.cpp

```diff
@@ -2669,6 +2669,97 @@ void inlineCharacterIsMethod(TR::Node *node, TR::MethodSymbol* methodSymbol, TR:
    cg->stopUsingRegister(tmpReg);
    }
 
+void buildCRC32CCall(TR::Node *callNode,
+               TR::RegisterDependencyConditions *deps,
+               TR::MethodSymbol* methodSymbol,
+               TR::CodeGenerator *cg,
+               TR::LabelSymbol *&returnLabel,
+               bool crc32m2, bool crc32m3)
+   {
+   TR::Compilation *comp = cg->comp();
+   uintptr_t targetAddress;
+   TR::Register        *gr2Reg;
+   bool aix_style_linkage = comp->target().isAIX() || (comp->target().is64Bit() && comp->target().isLinux());
+
+   if (aix_style_linkage)
+      {
+      gr2Reg = deps->searchPreConditionRegister(TR::RealRegister::gr2);
+      }
+
+   // Argument changes are needed
+   targetAddress = (uintptr_t)((comp->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P8) && comp->target().cpu.supportsFeature(OMR_FEATURE_PPC_HAS_VSX))?crc32_vpmsum:crc32_no_vpmsum);
+
+   // Assuming pre/postCondition have the same index, we use preCondition to map
+   OMR::RegisterDependencyMap map(deps->getPreConditions()->getRegisterDependency(0), deps->getAddCursorForPre());
+   for (int32_t cnt=0; cnt < deps->getAddCursorForPre(); cnt++)
+      map.addDependency(deps->getPreConditions()->getRegisterDependency(cnt), cnt);
+
+   TR::Register *addrArg, *posArg, *lenArg, *wasteArg;
+   if (crc32m2)
+      {
+      addrArg = map.getSourceWithTarget(TR::RealRegister::gr4);
+      posArg = map.getSourceWithTarget(TR::RealRegister::gr5);
+      lenArg = map.getSourceWithTarget(TR::RealRegister::gr6);
+
+      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi2, callNode, addrArg, addrArg, TR::Compiler->om.contiguousArrayHeaderSizeInBytes());
+      }
+
+   if (crc32m3)
+      {
+      addrArg = map.getSourceWithTarget(comp->target().is64Bit()?(TR::RealRegister::gr4):(TR::RealRegister::gr5));
+      posArg = map.getSourceWithTarget(comp->target().is64Bit()?(TR::RealRegister::gr5):(TR::RealRegister::gr6));
+      lenArg = map.getSourceWithTarget(comp->target().is64Bit()?(TR::RealRegister::gr6):(TR::RealRegister::gr7));
+      if (!comp->target().is64Bit())
+         wasteArg = map.getSourceWithTarget(TR::RealRegister::gr4);
+      }
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::add, callNode, addrArg, addrArg, posArg);
+
+   /* For CRC32C, java uses len arg as offset for crc calculation. To workaround this in vpmsum
+    * where (off + len) is passed as len, perform a sub operation to take offset from len
+    */
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::subf, callNode, lenArg, posArg, lenArg);
+   generateTrg1ImmInstruction(cg, TR::InstOpCode::li, callNode, posArg, 1);
```

Comment:
Let's add a comment here as well that tells the reader why `posArg` is being set to 1 and which parameter of the helper it is mapped to below.